hudi-common/src/main/java/org/apache/hudi/common/config/ConfigProperty.java

```diff
@@ -139,6 +144,49 @@ public ConfigProperty<T> withDocumentation(String doc) {
     return new ConfigProperty<>(key, defaultValue, docOnDefaultValue, doc, sinceVersion, deprecatedVersion, inferFunction, validValues, advanced, alternatives);
   }
 
+  public <U extends Enum<U>> ConfigProperty<T> withDocumentation(Class<U> e) {
+    return withDocumentation(e,"");
+  }
+
+  private <U extends Enum<U>> boolean isDefaultField(Class<U> e, Field f) {
+    if (!hasDefaultValue()) {
+      return false;
+    }
+    if (defaultValue() instanceof String) {
+      return f.getName().equals(defaultValue());
+    }
+    return Enum.valueOf(e, f.getName()).equals(defaultValue());
+  }
+
+  public <U extends Enum<U>> ConfigProperty<T> withDocumentation(Class<U> e, String doc) {
```

Comment:
Change this to `public <U extends Enum<U>> ConfigProperty<T> withDocumentation(Class<U> e)` and remove the following?
```
public <U extends Enum<U>> ConfigProperty<T> withDocumentation(Class<U> e) {
    return withDocumentation(e,"");
  }
```

Reply:
Why? Sometimes the config needs some extra explanation that the enum doesn't provide

Reply:
I see.  You mean the same enum may be used by multiple configs with different meanings like the file format for base and log files?